### PR TITLE
fix: fix configuration update racing condition

### PR DESCRIPTION
**Pull request by rickstaa**
This PR handles the racing condition that was introduced in #65 when the configuration object was updated (see https://github.com/Tom-Bonnike/vscode-formatting-toggle/pull/65#issuecomment-1186872006).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 .DS_Store
 node_modules/
 out/
+
+# Editor files
+.vscode
```
Comment (Tom-Bonnike) [inline .gitignore:6]: You usally want to commit the `.vscode` folder. Here we need `launch.json` checked in.
```suggestion
```

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+*.code-workspace
```
Comment (Tom-Bonnike) [inline .gitignore:7]: I’m not too precious about the `gitignore` file, but I’m just curious, isn‘t that only created if the repo has workspaces? 🤔

**Review by Tom-Bonnike — APPROVED** [review]
Looking good. I’ll just do some testing, merge your PR and release that patch 👀

File: .gitignore (hunk 2 above)
Comment (rickstaa) [reply .gitignore:6]: That is correct. Again mather of habbit. Feel free to remove it.

Comment (Tom-Bonnike) [reply .gitignore:6]: But workspace files would only be created in the “root” workspace repo nah? Never set one up 😄

**Comment by Tom-Bonnike** [comment]
Thanks Rick!

**Pull request merged**